fetch_telemetry.py

```diff
@@ -0,0 +1,316 @@
+    
+    #Dither
+    for msid in ['AODITHR2', 'AODITHR3']:
+        if msid in update_msids:
+            data[msid]['value'] = f"{float(data[msid]['value']) * 3600.0}"
+
+    #AC CCD Temperature
+    if 'AACCCDPT' in update_msids:
+        data['AACCCDPT']['value'] = f"{5.0 * (float(data['AACCCDPT']['value']) -32) / 9.0}" #---Convert F to C
+
+    #Battery SOC Range
+    for msid in ['EOCHRGB1', 'EOCHRGB2', 'EOCHRGB3']:      #--- percentage
+        if msid in update_msids:
+            data[msid]['value'] = f"{float(data[msid]['value']) * 100.0}"
+
+    return data
+
+
+def generate_psuedo_msids(data):
+    """
+    Create psuedo MSIDs for display
+    """
+    #Create "ACIS Stat7-0" msid
+    #Located in try-block so that if one of these values is missing from the blob update, then it is left alone to the last valid value
+    #Typically these will all be reported in the same blob.
+    try:
+        string = ''
+        time = 0
+        for msid in ['1STAT7ST', '1STAT6ST', '1STAT5ST', '1STAT4ST', '1STAT3ST', '1STAT2ST', '1STAT1ST', '1STAT0ST']:
+            if data[msid]['time'] > time:
+                time = data[msid]['time']
+            if float(data[msid]['value']) == 1:
+                string += 'T'
+            else:
+                string += 'F'
+        data['ACISSTAT'] = {'time': time, 'value': string}
+    except:
+        pass
+
+    #Compute ACA Fiducial
+    try:
+        string = ''
+        time = 0
+        for msid in ['AOACFID0', 'AOACFID1','AOACFID2','AOACFID3','AOACFID4','AOACFID5','AOACFID6','AOACFID7']:
+            if data[msid]['time'] > time:
+                time = data[msid]['time']
+            string += data[msid]['value'][0] # First letter in string
+        data['AOACFIDC'] = {'time': time, 'value': string}
+    except:
+        pass
+
+    #Compute ACA Image
+    try:
+        string = ''
+        time = 0
+        for msid in ['AOACFCT0', 'AOACFCT1','AOACFCT2','AOACFCT3','AOACFCT4','AOACFCT5','AOACFCT6','AOACFCT7']:
+            if data[msid]['time'] > time:
+                time = data[msid]['time']
+            string += data[msid]['value'][0] # First letter in string
+        data['AOACFCTC'] = {'time': time, 'value': string}
+    except:
+        pass
+
+    return data
+
+
+def check_limit_status(data):
+    """
+    Include the limit status into the data structure
+    """
+    for msid, entry in data.items():
+        status = cms.check_status(msid, entry['value'], LIMIT_DICT, data)
+        data[msid]['scheck'] = status
+    return data
+
+def update_json_blobs(data):
+    """
+    Iterate through blob_<part>.json updating each data value
+    """
+    for part in BLOB_SECTIONS:
+        with open(f"{HTML_DIR}/blob_{part}.json") as f:
+            data_list = json.load(f)
+#
+#--- Remove the dummy time entry
+#
+        for i in range(len(data_list)):
+            if data_list[i]['msid'] == "LASTDCHECK":
```

Comment:
We may not need LASTDCHECK anymore because with this optimized code The Last Data Update on top and LASTDCHECK should be very close to each other.

Reply:
Last Data Update depends on the LASTDCHECK time data entry, but the java script libraries split the entry into a separate string and then list that string differently compared to the LASTDCHECK value display. I can refactor this in another PR to make the calculation and display more direct.